contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -771,10 +771,10 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     ) private returns (TeleporterMessageReceipt[] memory result) {
         // Get the current outstanding receipts for the given chain ID.
         // If the queue contract doesn't exist, there are not outstanding receipts to send.
-        ReceiptQueue outstandingReceiptForDestination = outstandingReceipts[
+        ReceiptQueue.TeleporterMessageReceiptQueue storage outstandingReceiptForDestination = outstandingReceipts[
```

Comment:
Can we specify `outstandingReceiptForDestination` as `memory`? `storage` doesn't seem appropriate since this is a stack variable that doesn't need to be preserved between function calls.

Reply:
It unfortunately needs to be declared as `storage` since the struct contains a mapping, which can only live in storage of a contract.

That being said, could we move `_getOutstandingReceiptsToSend` to be a function in the `ReceiptQueue` library? I think that would likely be cleaner and safe on gas.

Reply:
move getOutstandingReceiptsToSend into the library.